src/main/java/duke/Duke.java

```diff
@@ -0,0 +1,159 @@
+package duke;
+
+import duke.Deadlines;
+import duke.Events;
+import duke.Task;
+import duke.ToDos;
+
+import java.util.Scanner;
+
+public class Duke {
+
+    public static void showWelcomeMessage() {
+        System.out.println("Hello! I'm duke.Duke");
+        System.out.println("What can I do for you?");
+    }
+
+    public static void printList(Task[] list, int size) {
+        int position = 1;
+        System.out.println("    Here are the tasks in your list:");
+        for( int i = 0 ; i < size ; i ++ ) {
+            System.out.println( "    " + position  + "." + list[i]);
+            position ++;
+        }
+    }
+
+    public static void markTaskDone(Task[] list , String word) {
+        //convert string number to int number
+        int position = Integer.parseInt(word);
+        //if task does not exist, return
+        if (position > list.length) {
+            System.out.println("    Sorry! No such task!");
+            return;
+        }
+        //if task exist, mark task as done
+        Task task = list[position - 1];
+        task.setDone();
+        //print notification
+        System.out.println("    Nice! I've marked this task as done:");
+        System.out.println("    " + task);
+    }
+
+    public static void addTask(Task[] list, Task newTask, int size) {
+        //add task to list
+        list[size] = newTask;
+        //print statements
+        System.out.println("    Got it. I've added this task:");
+        System.out.println("      " + list[size]);
+        System.out.println("    Now you have " + (size + 1) + " tasks in the list.");
+    }
+
+    public static Deadlines createNewDeadline(String[] words) {
+        int deadlineIndex = 0;
+        String by = "";
+        String deadlineDescription = "";
+        for(int i = 1 ; i < words.length ; i ++) {
+            if(words[i].equals("/by")) {
+                deadlineIndex = i + 1;
+                by = words[deadlineIndex];
+                break;
+            }
+            deadlineDescription = deadlineDescription + " " + words[i];
+        }
+        Deadlines newDeadline = new Deadlines(deadlineDescription.substring(1), by);
+        return newDeadline;
+    }
+
+    public static Events createNewEvent(String[] words) {
+        int eventIndex = 0;
+        String timeAllocation = "";
+        String eventDescription = "";
+        for(int i = 1 ; i < words.length ; i ++) {
+            if(words[i].equals("/at")) {
+                eventIndex = i + 1;
+                for(int j = eventIndex ; j < words.length ; j++) {
+                    timeAllocation = timeAllocation + " " + words[j];
+                }
+                break;
+            }
+            eventDescription = eventDescription + " " + words[i];
+        }
```

Comment:
Suggest to modify this code to avoid nested loop. You may want to look into Java String methods.